### PR TITLE
[core] [5/N] Check uv existence before installation

### PR DESCRIPTION
**Pull request by dentiny**
This PR checks uv existence before installation, if `uv` version not specified by the users.

### CODE REVIEW

File: python/ray/_private/runtime_env/uv.py [hunk 1]
```diff
@@ -62,6 +64,28 @@ async def _install_uv(
         logger.info("Installing package uv to %s", virtualenv_path)
         await check_output_cmd(uv_install_cmd, logger=logger, cwd=cwd, env=pip_env)
 
+    # TODO(hjiang): Add an integration test for existence check after
+    # PR (https://github.com/ray-project/ray/pull/48619) gets merged.
```
Comment (dentiny) [inline python/ray/_private/runtime_env/uv.py:68]: My planned way to test: 
- In integration test with `ray.remote` having `uv` specified, making sure all packages correctly installed;
- Call `_check_uv_existence` to make sure `uv` does exist in virtual env.

File: python/ray/_private/runtime_env/uv.py [hunk 2]
```diff
@@ -62,6 +64,28 @@ async def _install_uv(
...
+    async def _check_uv_existence(
+        self, path: str, cwd: str, env: dict, logger: logging.Logger
+    ) -> bool:
+        """Check and return the existence of `uv` in system executable path."""
```
Comment (jjyao) [inline python/ray/_private/runtime_env/uv.py:72]: This is checking the existence of `uv` inside the virtualenv we create for the runtime env right

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/runtime_env/uv.py (hunk 2 above)
Comment (dentiny) [reply python/ray/_private/runtime_env/uv.py:72]: Yes

File: python/ray/_private/runtime_env/uv.py [hunk 3]
```diff
@@ -91,6 +93,28 @@ async def _install_uv(
         logger.info("Installing package uv to %s", virtualenv_path)
         await check_output_cmd(uv_install_cmd, logger=logger, cwd=cwd, env=pip_env)
 
+    # TODO(hjiang): Add an integration test for existence check after
+    # PR (https://github.com/ray-project/ray/pull/48619) gets merged.
```
Comment (jjyao) [inline python/ray/_private/runtime_env/uv.py:97]: 48619 is merged now.

File: python/ray/_private/runtime_env/uv.py [hunk 4]
```diff
@@ -91,6 +93,28 @@ async def _install_uv(
...
+    async def _check_uv_existence(
+        self, path: str, cwd: str, env: dict, logger: logging.Logger
+    ) -> bool:
+        """Check and return the existence of `uv` in system executable path."""
```
Comment (jjyao) [inline python/ray/_private/runtime_env/uv.py:101]:
```suggestion
        """Check and return the existence of `uv` in virtualenv."""
```

Comment (dentiny) [reply python/ray/_private/runtime_env/uv.py]: Fixed.

**Pull request merged**